ruby/lib/gherkin3/token_scanner.rb

```diff
@@ -11,15 +13,23 @@ module Gherkin3
   # to look for Gherkin keywords for the associated language. The keywords are defined 
   # in gherkin-languages.json.
   class TokenScanner
+    protected
+
+    attr_reader :io
```

Comment:
> I also don't like that a String can mean file path.

I'm not sure if this breaks "your" API, but why not say a path needs to be a `Pathname`?

> The scanner is able to scan a String or IO. Nothing else (especially not another scanner).
> Maybe the whole if statement should move to the parse method. 

Why not using polymorphism for that?

* StringScanner
* IOScanner
* FileScanner

I will provide an implementation for that, then you can choose if you like it. :-)